### PR TITLE
[WIP] Get all passing Metal tests in CI [Bounty]

### PR DESCRIPTION
**Pull request by jacobpradels**
Todo list:

- [ ] Find all tests that pass on metal
- [ ] Enable them in CI

### CODE REVIEW

File: test/test_dtype.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ def test_bf16_disk_write_read(self):
 @unittest.skipIf((getenv("CI", "") != "" and Device.DEFAULT in ["LLVM"]) or Device.DEFAULT == "WEBGPU", "float16 broken in some CI backends")
 class TestHalfDtype(unittest.TestCase):
   def test_float16_to_np(self): _test_to_np(Tensor([1,2,3,4], dtype=dtypes.float16), np.float16, [1,2,3,4])
+  @unittest.skipIf(Device.DEFAULT == "METAL", "Metal Backend does not support float16")
```
Comment (wozeparrot) [inline test/test_dtype.py:88]: Metal does support float16???

**Comment by wozeparrot** [comment]
There has been quite a bit of investigation into this over at #1330, we are pretty sure that the main issue is a miscompilation somewhere, so there does need to be fix implemented there.

File: test/test_dtype.py (hunk 1 above)
Comment (wozeparrot) [reply test/test_dtype.py]: Yup that's the miscompilation I was talking about, check out #1330 for more info.

**Comment by jacobpradels** [comment]
> There has been quite a bit of investigation into this over at #1330, we are pretty sure that the main issue is a miscompilation somewhere, so there does need to be fix implemented there.

For the context of this PR do we want to continue skipping the test with a note to enable it when a fix is implemented or should this PR be blocked by the fix?

**Comment by wozeparrot** [comment]
I mean fixing the issue is a part of the bounty, so you can try to find a fix as well.

**Comment by jacobpradels** [comment]
@wozeparrot For clarification, what is the full scope of this bounty?  Is it to get every test that can run on metal to run on metal and pass?

Ex: the speed tests fail because tinygrad runs slower than pytorch on some operations (sub, sum).  Are optimizing those operations on metal part of this as well? This also includes a bunch of tests in `test_tensor.py`, but those tests don't seem to be ran on any of the back ends in the CI.

My original understanding was that this bounty was to implement all tests that currently pass for metal into the CI.

**Comment by jacobpradels** [comment]
Followed the story of this issue and understand the scope better. Closing PR.

**Pull request closed without merging**